node/src/components/api_server/rest_server.rs

```diff
@@ -0,0 +1,85 @@
+use futures::FutureExt;
+use http::Response;
+use hyper::Body;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use tracing::info;
+use warp::{
+    filters::BoxedFilter,
+    http::StatusCode,
+    reject::Rejection,
+    reply::{self, Reply},
+    Filter,
+};
+
+use super::{rpcs::info::GetStatusResult, ReactorEventT, CLIENT_API_VERSION};
+use crate::{
+    effect::{requests::ApiRequest, EffectBuilder},
+    reactor::QueueKind,
+};
+
+/// The status URL path.
+pub const STATUS_API_PATH: &str = "status";
+
+/// The metrics URL path.
+pub const METRICS_API_PATH: &str = "metrics";
+
+pub(super) fn create_status_filter<REv: ReactorEventT>(
+    effect_builder: EffectBuilder<REv>,
+) -> BoxedFilter<(Response<Body>,)> {
+    warp::get()
+        .and(warp::path(STATUS_API_PATH))
+        .and_then(move || {
+            effect_builder
+                .make_request(
+                    |responder| ApiRequest::GetStatus { responder },
+                    QueueKind::Api,
+                )
+                .map(|status_feed| {
+                    let body = GetStatusResult::from(status_feed);
+                    Ok::<_, Rejection>(reply::json(&body).into_response())
+                })
+        })
+        .boxed()
+}
+
+/// Response for a metrics request.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MetricsResponse {
+    /// The RPC API version.
+    pub api_version: Version,
+    /// The prometheus metrics as a string.
+    pub metrics: String,
+}
+
+pub(super) fn create_metrics_filter<REv: ReactorEventT>(
+    effect_builder: EffectBuilder<REv>,
+) -> BoxedFilter<(Response<Body>,)> {
+    warp::get()
+        .and(warp::path(METRICS_API_PATH))
+        .and_then(move || {
+            effect_builder
+                .make_request(
+                    |responder| ApiRequest::GetMetrics { responder },
+                    QueueKind::Api,
+                )
+                .map(|maybe_metrics| match maybe_metrics {
+                    Some(metrics) => {
+                        let body = MetricsResponse {
+                            api_version: CLIENT_API_VERSION.clone(),
+                            metrics,
+                        };
+                        Ok::<_, Rejection>(reply::json(&body).into_response())
```

Comment:
Hmm, JSON? I thought that Prometheus expects just plain text.

Reply:
I wanted to have all messages from the server contain the API version.  Maybe that's not reasonable here.  Is there a good way to inject the version into the metrics string, or is that likely wasted effort if Prometheus will ignore it anyway?

Either way - good catch.  I'll fix this to return a plain text response.

Reply:
I am not sure actually what Prometheus does with entries that it cannot parse.